### ISSUE TITLE
Targeted message: use `AuthPSKEncryptWithLabel`?

### ISSUE THREAD

**Issue by TWal** [mlswg/mls-extensions#16]
To be more future-proof, similarly to `EncryptWithLabel`, I guess we could design a similar function for the HPKE AuthPSK mode?
This would help ensuring future extensions that would be using the AuthPSK mode with leaf HPKE keys don't interfere with each other.

**Comment by TWal** [comment]
It looks like this point has been addressed with the safe extensions API. However, Targeted Messages re-define a safe API for SealAuthPSK, this could be moved in the safe extension API.

**Comment by rohanmahy** [comment]
I'm not going to comment about if the specific construction is safe or not, however TargetedMessage is now a true extension. It's not supposed to use the Safe Applications API because it isn't a component. It does need to use a reasonable set of labels, but these can be hardcoded and IANA registered.

**Comment by raphaelrobert** [comment]
> It looks like this point has been addressed with the safe extensions API. However, Targeted Messages re-define a safe API for SealAuthPSK, this could be moved in the safe extension API.

The issue with that proposal is that the safe HPKE API uses `EncryptWithLabel` from RFC9420 that uses `SealBase` and there is no counterpart for `SealAuthPSK`. If there was, we could probably make targeted messages a component rather than an extension.

**Comment by rohanmahy** [comment]
@raphaelrobert 
>we could probably make targeted messages a component rather than an extension.

No. under the current definition, there is no way that a new proposal or a new wireformat can be a component. That's why there is no subsection on them under Section 4. So it is really no loss.

As Richard mentioned, if we make a generic `EncryptWithAuthPSKLabel`, it will not work with PQ ciphersuites, so that's unfortunate.
